projects/plugins/boost/app/modules/cache/Boost_Cache.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+namespace Automattic\Jetpack_Boost\Modules\Page_Cache;
+
+abstract class Boost_Cache {
+	/*
+	 * @var string - The path key used to identify the cache directory for the current request. MD5 of the request uri.
+	 */
+	protected $path_key = false;
+
+	/*
+	 * @var string - The sanitized path for the current request.
+	 */
+	protected $request_uri = false;
+
+	public function __construct() {
+		$this->request_uri = isset( $_SERVER['REQUEST_URI'] )
+			? $this->sanitize_request_uri( $_SERVER['REQUEST_URI'] ) // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+			: false;
+	}
+
+	/*
+	 * Returns true if the request is cacheable.
+	 *
+	 * If a request is in the backend, or is a POST request, or is not an
+	 * html request, it is not cacheable.
+	 * The filter boost_cache_cacheable can be used to override this.
+	 *
+	 * @return bool
+	 */
+	public function is_cacheable() {
+		if ( ! apply_filters( 'boost_cache_cacheable', $this->request_uri ) ) {
+			return false;
+		}
+
+		if ( $this->is_backend() ) {
+			return false;
+		}
+
+		if ( isset( $_SERVER['REQUEST_METHOD'] ) && $_SERVER['REQUEST_METHOD'] === 'POST' ) {
+			return false;
+		}
+
+		if ( isset( $_SERVER['HTTP_ACCEPT'] ) && strpos( $_SERVER['HTTP_ACCEPT'], 'text/html' ) === false ) { // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+			return false;
+		}
+
+		return true;
+	}
+
+	/*
+	 * copy of _deep_replace() to be used before WordPress loads
+	 * @see https://developer.wordpress.org/reference/functions/_deep_replace/
+	 */
+	private function deep_replace( $search, $subject ) {
+		$subject = (string) $subject;
+
+		$count = 1;
+		while ( $count ) {
+			$subject = str_replace( $search, '', $subject, $count );
+		}
+
+		return $subject;
+	}
+
+	/*
+	 * Sanitize the request uri so it can be used for caching purposes.
+	 * It removes the query string and the trailing slash, and characters
+	 * that might cause problems with the filesystem.
+	 *
+	 * @param string $request_uri - The request uri to sanitize.
+	 * @return string - The sanitized request uri.
+	 */
+	protected function sanitize_request_uri( $request_uri ) {
+		// get path from request uri
+		$request_uri = parse_url( $request_uri, PHP_URL_PATH ); // phpcs:ignore WordPress.WP.AlternativeFunctions.parse_url_parse_url
+		if ( $request_uri === '' ) {
+			$request_uri = '/';
+		} elseif ( substr( $request_uri, -1 ) !== '/' ) {
+			$request_uri .= '/';
+		}
+
+		$request_uri = $this->deep_replace(
+			array( '..', '\\', 'index.php' ),
```

Comment:
I could probably leave `index.php` out of that list (as it'll be removed by parse_url anyway), but we need the deep_replace to make sure ".." doesn't end up in the final path, which would allow an attacker to perform a directory traversal. "\" doesn't belong in URLs either, and might be used to escape a character.

Reply:
I removed the entire deep_replace() code and renamed the function "normalize_request_uri", with dire comments about using the variable. :)